frontend/packages/process-editor/src/components/Canvas/Canvas.test.tsx

```diff
@@ -0,0 +1,57 @@
+import React  from 'react';
+import { Viewer } from './Canvas';
+import { render , screen } from '@testing-library/react';
+import BpmnJs from 'bpmn-js/dist/bpmn-navigated-viewer.production.min';
+
+const renderNoDiagramError = jest.fn();
+const renderNoProcessError = jest.fn(); 
+
+const mockBpmn = {
+  get: jest.fn().mockReturnValue({
+    zoom: jest.fn(),
+    stepZoom: jest.fn(),
+    add: jest.fn(),
+    onSave: jest.fn(),
+  }),
+  destroy: jest.fn(),
+  on: jest.fn(),
+  importXML: jest.fn(),
+};    
+
+jest.mock('bpmn-js/dist/bpmn-navigated-viewer.production.min')
+  BpmnJs.mockImplementation(() => mockBpmn);
+
+jest.mock('../../hooks/useBpmnViewer', () => ({
+    useBpmnViewer: jest.fn(() => ({
+      renderNoDiagramError: true, 
+      renderNoProcessError: true, 
+      canvasRef: jest.fn(),
+    })),
+  }));
+
+  describe('Viewer Component', () => {
+    it('renders the Alert component when there is no diagram', () => {
+      render(<Viewer />);
+      expect(screen.getByTestId("no_diagram")).toBeInTheDocument();
```

Comment:
Here we should use the getByRole :)
```suggestion
      expect(screen.getByRole('heading', { name: textMock('key'), level: 1 }));
```

Reply:
I suggest that we remove all the test-id within this unit-test. :)